src/Layout/index.js

```diff
@@ -16,13 +17,24 @@ class Layout extends Component {
       window.GA_INITIALIZED = true
     }
     logPageView()
+
+    Router.events.on('routeChangeComplete', this.handleRouteChange)
+  }
+
+  componentWillUnmount() {
+    Router.events.off('routeChangeComplete', this.handleRouteChange)
   }
 
   UNSAFE_componentWillReceiveProps(nextProps) {
     const { router } = nextProps
     this.isDocPage = router.pathname.split('/')[1] === 'doc'
   }
 
+  handleRouteChange = () => {
+    // Without timeout GA will send old page title
```

Comment:
is it a recommended way to prevent this? looks fragile a bit.

Reply:
can we just provide an argument to the logPageView?

Reply:
I refactored layout a little, and replaced timeout with `requestAnimationFrame`, but I afraid we still need it. Basically Next's `Head` component uses promise inside (https://github.com/zeit/next.js/blob/18a9c7e371efc4c487f9c3599c3211ce30009d6c/packages/next/client/head-manager.js#L14), and, because of that, page content renders and `componentDidUpdate` fires before the `<head>` content is updated. Adding rAF there will add GA task to the next loop of Event Loop after the head update.